README.md

```diff
@@ -1,117 +1,250 @@
 
-```json
-{
-    "ServiceSettings": {
-        ...
-        "EnableLocalMode": true,
-        "LocalModeSocketLocation": "/var/tmp/mattermost_local.socket"
-    }
-}
-```
+#### Receiving notifications about changes to the configuration
 
-and then deploy your plugin:
-```
-make deploy
-```
+![Screenshot of receiving notifications about changes to the configuration in action](screenshots/changes.png)
 
-You may also customize the Unix socket path:
-```
-export MM_LOCALSOCKETPATH=/var/tmp/alternate_local.socket
-make deploy
-```
+### Audience
 
-If developing a plugin with a webapp, watch for changes and deploy those automatically:
-```
-export MM_SERVICESETTINGS_SITEURL=http://localhost:8065
-export MM_ADMIN_TOKEN=j44acwd8obn78cdcx7koid4jkr
-make watch
-```
+This guide is for Mattermost System Admins setting up the Nagios plugin and Mattermost users who want information about the plugin functionality.
 
-### Deploying with credentials
+### Important notice
 
-Alternatively, you can authenticate with the server's API with credentials:
-```
-export MM_SERVICESETTINGS_SITEURL=http://localhost:8065
-export MM_ADMIN_USERNAME=admin
-export MM_ADMIN_PASSWORD=password
-make deploy
-```
+If you are a Nagios admin/user and think there is something this plugin lacks or something that it does could be done the other way around, let us know!
+We are trying to develop this plugin based on users' needs.
+If there is a certain feature you or your team needs, open up an issue, and explain your needs.
+We will be happy to help.
+
+## Installing the plugin
+
+1. Download the latest stable version of the plugin from the [releases page](https://github.com/ulumuri/mattermost-plugin-nagios/releases)
+2. In Mattermost, go to **System Console → Plugins → Management**
+3. Upload the plugin in the **Upload Plugin** section
+4. Configure the plugin before you enable it :arrow_down:
 
-or with a [personal access token](https://docs.mattermost.com/developer/personal-access-tokens.html):
+## Configuring the plugin
+
+1. Enter the URL for your Nagios instance
+    1. In Mattermost, go to **System Console → Plugins → Nagios**
+    2. Set the **Nagios URL**
+        1. Remember to add `http://` or `https://` at the beginning!
+        2. Example: `https://nagios.fedoraproject.org`
+2. Click *Save* to save the settings
+3. In Mattermost, go to **System Console → Plugins → Management** and click *Enable* underneath the Nagios plugin
+4. The plugin is now ready to use! :congratulations:
+
+### Configuring the configuration files watcher
+
+*This step is optional, although highly recommended.*
+
+1. Regenerate the token for the configuration files watcher
+    1. In Mattermost, go to **System Console → Plugins → Nagios**
+    2. Click *Regenerate* to regenerate the token
+    3. Copy the token (you are going to use it later)
+2. Click *Save* to save the settings
+3. Switch to the machine where Nagios is running (preferably)
```

Comment:
This looks invasive, does nagios API provide similar functionality?

Reply:
Unfortunately not. :( The Nagios API does not provide such functionality whatsoever. It's even worse: Nagios have no idea its configuration has changed while it's running. The only way to reload the configuration is to restart it.

I agree this looks invasive. I made this point on the Plugin: Nagios channel ([link to the discussion](https://community.mattermost.com/core/pl/r3rgadinzjf4xdqpgsjzu87xah)), and we have decided it's the best way to do this monitoring thing right now.